File: spec/support/factories/characters.rb
FactoryBot.define do
  factory :character do
    association :user
    name { "Arannis" }
    character_class { "Wizard" }
    subclass { "School of Evocation" }
    species { "Elf" }
    level { 1 }
    strength { rand(8..20) }
    dexterity { rand(8..20) }
    constitution { rand(8..20) }
    intelligence { rand(8..20) }
    wisdom { rand(8..20) }
    charisma { rand(8..20) }
    hitpoints { rand(8..12) }
    speed { 30 }

    trait :warrior do
      character_class { "Warrior" }
      subclass { "Path of the Berserker" }
    end

    trait :mage do
      character_class { "Mage" }
      subclass { "College of Lore" }
    end

    trait :guardian do
      character_class { "Guardian" }
      subclass { "Life Domain" }
    end

    trait :mystic do
      character_class { "Mystic" }
      subclass { "Circle of Nature" }
    end

    trait :scout do
      character_class { "Scout" }
      subclass { "Pathfinder" }
    end

    trait :rogue do
      character_class { "Rogue" }
      subclass { "Thief" }
    end

    trait :sage do
      character_class { "Sage" }
      subclass { "School of Knowledge" }
    end

    trait :ranger do
      character_class { "Ranger" }
      subclass { "Hunter" }
    end

    trait :warlock do
      character_class { "Warlock" }
      subclass { "Pact of the Fey" }
    end

    trait :wizard do
      character_class { "Wizard" }
      subclass { "School of Evocation" }
    end

    trait :artificer do
      character_class { "Artificer" }
      subclass { "Alchemist" }
    end
  end
end

File: app/assets/javascripts/characters.js
document.addEventListener("DOMContentLoaded", () => {
  const classSelect = document.getElementById("character_class_select");
  const subclassSelect = document.getElementById("subclass_select");

  const subclassOptions = {
    "Warrior": ["Path of the Berserker", "Path of the Guardian", "Path of the Champion"],
    "Mage": ["College of Lore", "College of Battle", "College of Elements"],
    "Guardian": ["Life Domain", "Protection Domain", "Justice Domain"],
    "Mystic": ["Circle of Nature", "Circle of Spirits", "Circle of Elements"],
    "Scout": ["Pathfinder", "Beast Master", "Shadow Walker"],
    "Rogue": ["Thief", "Assassin", "Trickster"],
    "Sage": ["School of Knowledge", "School of Divination", "School of Illusion"],
    "Ranger": ["Hunter", "Beast Companion", "Wilderness Guide"],
    "Warlock": ["Pact of the Fey", "Pact of the Void", "Pact of the Elements"],
    "Wizard": ["School of Evocation", "School of Abjuration", "School of Conjuration"],
    "Artificer": ["Alchemist", "Engineer", "Enchanter"]
  };

  if (!classSelect || !subclassSelect) return;

  classSelect.addEventListener("change", () => {
    const selectedClass = classSelect.value;
    const subclasses = subclassOptions[selectedClass] || [];

    // Clear old options
    subclassSelect.innerHTML = "";

    if (subclasses.length > 0) {
      subclassSelect.disabled = false;
      const promptOption = document.createElement("option");
      promptOption.text = "Choose a subclass";
      promptOption.value = "";
      subclassSelect.add(promptOption);

      subclasses.forEach(subclass => {
        const option = document.createElement("option");
        option.text = subclass;
        option.value = subclass;
        subclassSelect.add(option);
      });
    } else {
      subclassSelect.disabled = true;
    }
  });
});
